argument.go

```diff
@@ -42,18 +42,21 @@ func (o arg) GetLname() string {
 
 type help struct{}
 
+// HelpCalled error returned when parsing a help argument
+// instead of exiting to prevent further parsing
+type HelpCalled struct {
+	arg arg
+}
+
+func (hc *HelpCalled) Error() string {
+	return fmt.Sprintf("Command %s help argument parsed", hc.arg.parent.name)
+}
+
 //Check if argumet present.
 //For args with size 1 (Flag,FlagCounter) multiple shorthand in one argument are allowed,
 //so check - returns the number of occurrences.
 //For other args check - returns 1 if occured or 0 in no
 func (o *arg) check(argument string) int {
-	// Shortcut to showing help
-	if argument == "-h" || argument == "--help" {
-		helpText := o.parent.Help(nil)
-		fmt.Print(helpText)
-		os.Exit(0)
-	}
-
```

Comment:
This is here so that when `-h|--help` is present, it would terminate further parsing and produce help message. Showing help message should precede any other arguments (so if someone uses invalid args it won't show error and instead will show help message).

I am not sure how this new implementation would do that, is this expectation still correct?

Reply:
I removed the help argument section in check because it was redundant and check doesn't provide an error return type for the HelpCalled struct. The only place check is called is:
https://github.com/akamensky/argparse/blob/543653f1125f6a93550d420965ac435a0efa5dcb/command.go#L91-L107
Perhaps the check call could be moved into parse, since that seems to be where it belongs?